apis/core/v1alpha2/readiness_types.go

```diff
@@ -7,22 +7,48 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
-// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
-
 // ReadinessSpec defines the desired state of Readiness
 type ReadinessSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// Checks is the set of checks that are required to mark the readiness
+	Checks []Check `json:"checks"`
+}
+
+type Check struct {
+	// Name is the name of the check
+	Name string `json:"name"`
 
-	// Foo is an example field of Readiness. Edit readiness_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// Type is the type of the check. the Type can be either basic or composite
```

Comment:
A brief description about what these enums indicate can be added here.

Reply:
Done

Reply:
There are few fields that has to be added to support the composite type. But, have not added them now. We can add those later.